distribution/push_v2.go

```diff
@@ -199,6 +199,15 @@ func (p *v2Pusher) pushV2Tag(ctx context.Context, ref reference.NamedTagged, ima
 
 	manifestDigest := digest.FromBytes(canonicalManifest)
 	progress.Messagef(p.config.ProgressOutput, "", "%s: digest: %s size: %d", ref.Tag(), manifestDigest, len(canonicalManifest))
+
+	// Best effort store the digest
+	digest, err := reference.WithDigest(ref, manifestDigest)
+	if err != nil {
+		logrus.Errorf("Failed to store digest reference %s: %s", manifestDigest, err.Error())
```

Comment:
Same as previous comment, the `reference.WithDigest` call is only making the digest reference and should not fail. The error to be logged is on `AddDigest`